validphys2/src/validphys/fkparser.py

```diff
@@ -172,56 +192,52 @@ def _parse_hadronic_fast_kernel(f):
     # Note that we need the slower whitespace here because it turns out
     # that there are fktables where space and tab are used as separators
     # within the same table.
-    df = pd.read_csv(f, sep=r'\s+', header=None, index_col=(0,1,2))
-    df.columns = list(range(14*14))
-    df.index.names = ['data', 'x1', 'x2']
+    df = pd.read_csv(f, sep=r"\s+", header=None, index_col=(0, 1, 2))
+    df.columns = list(range(14 * 14))
+    df.index.names = ["data", "x1", "x2"]
     return df
 
+
 def _parse_dis_fast_kernel(f):
     """Parse the FastKernel section of a DIS FKTable into a DataFrame.
     ``f`` should be a stream containing only the section"""
-    df = pd.read_csv(f, sep=r'\s+', header=None, index_col=(0,1))
+    df = pd.read_csv(f, sep=r"\s+", header=None, index_col=(0, 1))
     df.columns = list(range(14))
-    df.index.names = ['data', 'x']
+    df.index.names = ["data", "x"]
     return df
 
 
 def _parse_gridinfo(line_and_stream):
     dict_result, line_number, next_line = _parse_fk_options(
-        line_and_stream,
-        value_parsers={
-            "HADRONIC": _bytes_to_bool,
-            "NDATA": int,
-            "NX": int
-        })
+        line_and_stream, value_parsers={"HADRONIC": _bytes_to_bool, "NDATA": int, "NX": int}
+    )
     gi = GridInfo(**{k.lower(): v for k, v in dict_result.items()})
     return gi, line_number, next_line
 
 
-
 def _parse_header(lineno, header):
     if not _is_header_line(header):
-        raise BadFKTableError(f"Bad header at line {lineno}: First "
-                "character should be either '_' or '{'")
+        raise BadFKTableError(
+            f"Bad header at line {lineno}: First " "character should be either '_' or '{'"
```

Comment:
Uh, sorry but this is a bit too much reformatting. I find many of the changes (e.g. in the spaces between multiplications) for the worse. And what it is doing here is senseless. So I am going to quote

> This tool should not be used to aggressively reformat existing files.

https://docs.nnpdf.science/contributing/python-tools.html?highlight=black#python-static-checks-and-code-style

Reply:
I rather fix manually the problems that black might have created, `fkparser` is a file where almost all that I haven't touched explicitly I either want to remove or will remove eventually so it doesn't make sense not to reformat for the sake of some decisions taken 5 years ago.

Reply:
Seeing as I am the one who wrote the code, seeing the formatting bulldozed makes me somewhat irrationally unhappy. But on top of that the facts that it does obviously silly things as in here and that makes the diff much harder to read are the reasons for the "don't aggressively reformat" bit in the docs, and which I'd argue are valid.

Also I am not sure I'd be comfortable removing this code soon. We would need that all the infrastructure and all the theories are working with the new grids and iron out all the problems that might appear with more widespread and prolonged usage .

Reply:
> Also I am not sure I'd be comfortable removing this code soon. We would need that all the infrastructure and all the theories are working with the new grids and iron out all the problems that might appear with more widespread and prolonged usage .

Yes, but it should become "legacy" after this PR. Honestly I would like to regenerate all theories as soon as all issues are ironed out but that might just be a wish.

Reply:
I guess that depends on what you mean with "all theories": all that are in the db? that might be too much - all that are actually used (which I estimated at ~25 - with the biggest chunk being SV, which we have to regenerate in any case)? maybe yes

Reply:
Why is it too much? If it can be done automatically we can just leave it running for two months if needed and at the end we have our new shiny fktables. Or are some for which this will not be doable?

Reply:
I think there are things that just require some time to be ascertained. Even if we compare point by point all fktables in all theories (which btw we have no ability to produce with the new toolchain at the moment) and they are the same today I would still want to see that things remain stable for some iterations of the new toolchain. 

For one the existing fktable format is easy enough to parse by hand and has been stable for some 9 years now  while the new one cannot be parsed without the implementation and might be non trivial to get to work between updates of pinnaple.  While there are ways to deal with format upgrades, having some alternative source of truth to compare with might be helpful for a while.

Lastly I don't think maintaining the two formats should require too much work (as for the time being it is just a matter of not deleting the existing fkparser) seeing as the two things end up in the same memory object at the time. This is different from say the cost of dealing with c++.

Reply:
> Why is it too much? If it can be done automatically we can just leave it running for two months if needed and at the end we have our new shiny fktables. Or are there some for which this will not be doable?

no no, technically there is no problem - it's just a matter of resources ...